src/lib/list/list.scss

```diff
@@ -43,6 +43,19 @@ $mat-dense-list-icon-size: 20px;
     position: relative;
   }
 
+  .mat-list-item-content-reverse {
+    display: flex;
+    //flex-direction: row;
+    align-items: center;
+    box-sizing: border-box;
+    height: $base-height;
+    padding: 0 $mat-list-side-padding;
+    position: relative;
+    flex-direction: row-reverse;
+    -webkit-justify-content: space-around; /* Safari */
```

Comment:
You shouldn't need the `-webkit-justify-content`, since Safari 9 and 10 both support all of the flex properties unprefixed.